src/dss/utils.py

```diff
@@ -32,58 +32,88 @@ def __init__(self, msg: str, *args):
         self.msg = str(msg)
 
 
+def node_has_gpu_labels(lightkube_client: Client, labels: List[str]) -> bool:
```

Comment:
This asserts that all nodes have the provided labels, not that a given node has gpu labels.

Alternatively you could include the gpu labels here in the function code drop the `labels` input

```suggestion
def all_nodes_have_labels(lightkube_client: Client, labels: List[str]) -> bool:
```